src/common.speech/ServiceRecognizerBase.ts

```diff
@@ -349,12 +349,12 @@ export abstract class ServiceRecognizerBase implements IDisposable {
                                 if (!!this.privRecognizer.speechEndDetected) {
                                     this.privRecognizer.speechEndDetected(this.privRecognizer, speechStopEventArgs);
                                 }
-
+                                break;
+                            case "turn.end":
                                 if (requestSession.isSpeechEnded && this.privRecognizerConfig.isContinuousRecognition) {
```

Comment:
i changed the code to be more consistent with the c++ implementation, i.e., make sure we fire the event in case the stream has ended and we have seen a turn-start message.
in this case, we fire the eof cancel event once.

re-buffering on the audio side is done through the audio-node, so we are not loosing data here, however,
since we are not yet handling the timestamps in the reco messages, it might be that the message come "too early" which means, there is so much data in flight that there will be a subsequent turn.start/end after the current one.
(note: this /should/ only happen for file data since microphone data is streamed in realtime to the service and thus responsed should not be behind the actual audio data)